src/stepFunctions/asl/asl-yaml-languageservice.ts

```diff
@@ -0,0 +1,245 @@
+        } else if (document.getText().substring(offset - 2, offset) === ': ') {
+            // yaml doesn't allow auto-completion from white-space after certain nodes
+            atSpace = true
+            // initialize an empty string and adjust position to within the string before parsing yaml to json
+            const newText = document.getText().substring(0, offset) + '""\r' + document.getText().substr(offset)
+            yamldoc = parseYAML(newText)
+            currentDoc = matchOffsetToDocument(offset, yamldoc)
+            position.character += 1
+        } else if (node.type === 'property') {
+            // allow auto-completion of States field from empty space
+            if (node.keyNode.value === 'States') {
+                if (document.getText().substring(offset - 2, offset) === '  ') {
+                    // initialize brackets to surround the empty space when parsing
+                    const newText = document.getText().substring(0, offset) + '{}\r' + document.getText().substr(offset)
+                    yamldoc = parseYAML(newText)
+                    currentDoc = matchOffsetToDocument(offset, yamldoc)
+                    position.character += 1
+                } else {
+                    position.character -= 1
+                }
+            }
+        }
+
+        const aslCompletions = doCompleteAsl(document, position, currentDoc, yamlCompletions, {
+            ignoreColonOffset: true,
+        })
+
+        aslCompletions.items.forEach(completion => {
+            // format json completions for yaml
+            if (completion.textEdit) {
+                // textEdit can't be done on white-space so insert text is used instead
+                if (atSpace) {
+                    completion.insertText = completion.textEdit.newText
+                    // remove any commas from json-completions
+                    completion.insertText = completion.textEdit.newText.replace(/[\,]/g, '')
+                    completion.textEdit = undefined
+                } else {
+                    completion.textEdit.range.start.character = position.character
+                }
+            } else {
+                // remove null value auto-completions for 'string' nodes that don't support null values
+                completion.insertText = completion?.insertText?.replace(/\$\{1:null\}/g, '')
+            }
+        })
+        return Promise.resolve(aslCompletions)
+    }
+
+    languageService.doResolve = function(item: CompletionItem): Thenable<CompletionItem> {
+        return aslLanguageService.doResolve(item)
+    }
+
+    languageService.doHover = function(
+        document: TextDocument,
+        position: Position,
+        jsonDocument: JSONDocument
+    ): Thenable<Hover | null> {
+        const doc = parseYAML(document.getText())
+        const offset = document.offsetAt(position)
+        const currentDoc = matchOffsetToDocument(offset, doc)
+        if (currentDoc === null) {
+            return null!
```

Comment:
I did that to temporarily resolve the issue of the function needing a strictly Thenable < Hover[] || null> return type. I will change it to a Promise.resolve(null) since it's more appropriate